lib/spack/spack/test/util/executable.py

```diff
@@ -40,6 +40,36 @@ def test_read_unicode(tmpdir, working_env):
         assert u'\xc3' == script(output=str).strip()
 
 
+def test_which_relative_path_with_slash(tmpdir, working_env):
+    tmpdir.ensure('exe')
+    path = str(tmpdir.join('exe'))
+    os.environ['PATH'] = ''
+
+    with tmpdir.as_cwd():
+        no_exe = ex.which('./exe')
+        assert not no_exe
```

Comment:
Just a minor OCD, but can we change this to the functionally equivalent:
```suggestion
        exe = ex.which('./exe')
        assert not exe
```
or
```suggestion
        no_exe = ex.which('./exe')
        assert no_exe is None
```
The double negation in `not no_exe` feels weird to me.